client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import noop from 'lodash/noop';
+import i18n from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import Spinner from 'components/spinner';
+import { getCurrentUserEmail } from 'state/current-user/selectors';
+import {
+	verifyEmail,
+	resetVerifyEmailState,
+} from 'state/current-user/email-verification/actions';
+
+class VerifyEmailDialog extends Component {
+
+	getResendButtonLabel() {
+		if ( this.props.emailSent || this.props.error ) {
+			return i18n.translate( 'Email Sent' );
```

Comment:
this _may_ be a logical error. _if_ the request failed then we shouldn't anticipate that WordPress.com servers actually sent out the email, right, but yet here we indicate that it did?